src/main/java/com/ericsson/gerrit/plugins/eiffel/events/generators/EiffelSourceChangeCreatedEventGenerator.java

```diff
@@ -80,21 +75,11 @@ public static EiffelSourceChangeCreatedEvent generate(final EiffelPluginConfigur
         eiffelEvent.eventParams.data.gitIdentifier.branch = branch;
         eiffelEvent.eventParams.data.gitIdentifier.repoName = projectName;
 
-        final String previousSourceChangeCreatedEvent = getPreviousEiffelEventId(pluginConfig,
-                EventType.SCC_EVENT, changeId);
-        final Link previousVersionLink = createLink(LINK_TYPE_PREVIOUS_VERSION,
-                previousSourceChangeCreatedEvent);
-        if (previousVersionLink != null) {
-            eiffelEvent.eventParams.links.add(previousVersionLink);
-        }
+        final LinkGenerator linkGenerator = new LinkGenerator(pluginConfig, commitInformation);
+        linkGenerator.addSccBase(commitId);
```

Comment:
can't we move this logic to LinkGenerator directly? without having to explicitly call the addSccBase and addSccPreviousVersion?

Reply:
The links are different for the two events and I thought it was good to let that logic be part of the generator classes

Reply:
ok.